Boards/nucleo f746zg

hugueslarrive opened the pull request:
### Contribution description
- dma, eth, and spi features added (inspired by those of the f767zi which very similar board)
- configuration of the dma added for the spi
- pwm feature added

spi_dma and pwm could be backported to the 767zi but I haven't one to make tests.

### Testing procedure
- tests/gnrc_udp : ifconfig and ping to my linux host ok, flood ping from my linux host ok
- tests/periph_spi : bench ok at all clock speed for the two devices
- tests/periph_spi_dma : ok except for the "3" clock speed on device 1, "3" and "4" clock speed on device 0 depending which one is used after the other (the first one work, the second don't), I have tried to set another speed between those two but that is not help... I need some help to solve those bugs.
- tests/pwm : saw the duty-cycle oscillation on the 8 outputs configured on the scope

### Issues/PRs references

<!--
Examples: Fixes #1234. See also #5678. Depends on PR #9876.

Please use keywords (e.g., fixes, resolve) with the links to the issues you
resolved, this way they will be automatically closed when your pull request
is merged. See https://help.github.com/articles/closing-issues-using-keywords/.
-->

benpicco commented:
Nice!
We are transitioning to Kconfig, so the board's `Kconfig` file needs to be updated with the new features too.

hugueslarrive commented:
I updated my fork to add and update Kconfig which also fixed spi_dma bugs !

Now tests/periph_spi_dma bench is ok at all the clocks speed for both devices.

hugueslarrive commented:
It was a shame to be limited to a clock speed of 6.75MHz on a device that can go up to 50MHz and also to have twice the same speed of 421KHz for APB2.

I have selected identical speeds for APB1 for better consistency in case someone wants to configure SPI2 or SPI3.

aabadie commented on boards/nucleo-f746zg/include/periph_conf.h:
```diff
@@ -72,23 +131,129 @@ static const uart_conf_t uart_config[] = {
 
+/**
+ * @name   SPI configuration
+ *
```
```suggestion
```

aabadie commented on boards/nucleo-f746zg/include/periph_conf.h:
```diff
@@ -72,23 +131,129 @@ static const uart_conf_t uart_config[] = {
+ *
+ * @{
+ */
+static const uint8_t spi_divtable[2][5] = {
+    {       /* for APB1 @ 54000000Hz */
+        6,  /* -> 421875Hz */
+        5,  /* -> 843750Hz */
+        4,  /* -> 1687500Hz */
+        2,  /* -> 6750000Hz */
+        1   /* -> 13500000Hz */
+    },
+    {       /* for APB2 @ 108000000Hz */
+        7,  /* -> 421875Hz */
+        6,  /* -> 843750Hz */
+        5,  /* -> 1687500Hz */
+        3,  /* -> 6750000Hz */
+        2   /* -> 13500000Hz */
+    }
+};
+
```
You can remove the divtable declaration and use the common define in https://github.com/RIOT-OS/RIOT/blob/master/boards/common/stm32/include/cfg_spi_divtable.h because it already defines the same values.

Just add `#include "cfg_spi_divtable.h"` under https://github.com/RIOT-OS/RIOT/pull/14731/files#diff-c93eeec96927441066fd1246376c06e4R23.

```suggestion
```

hugueslarrive replied:
Yes but no, that is questionable : cfg_spi_divtable.h defines 210, 421, 843, 3375 and 6750KHz for APB1 and twice 421, 843, 3375 an 6750KHz for APB2 which is inconsistent for the lowest speed and it is a shame to be limited to 6,75MHz on a 50MHz capable device as I already explained in my previous comment.

So I understand your request on the principle for the homogeneity of the system, backward compatibility, bla bla... although I think that the auto-generation algorithm of spi_divtable is stupid. Sorry.

Even if they had the same values, they are so far different from this MCU possibilities that I think it is better to define them in periph_conf.h to ease their customization.

In our project we previously used a much less powerful atmega328p with an 8MHz SPI clock speed so you can hear that the auto-generated spi_divtable isn't usable in the real world.

Finally, be free to do it yourself if you really want, I am free to carry on using EXTERNAL BOARD_DIRS, that's the game !

bergzand replied:
Hello @hugueslarrive.

[I've fiddled around a bit](https://github.com/bergzand/RIOT/tree/pr/stm32/dynamic_spi_freqs) with the stm32 SPI implementation to get support for arbitrary SPI clock speeds. This completely removes the need for the `spi_divtable` and allows for supplying arbitrary clock speeds as `spi_clock_t` param. Would this solve your issue with the current SPI implementation on the stm32 platform?

hugueslarrive replied:
With this patch and tests/periph_spi I get:
- init 0 0 0 3 4: 1687500Hz
- init 0 0 1 3 4: 54000000Hz
- init 0 0 2 3 4: 843750Hz
- init 0 0 3 3 4: 3375000Hz
- init 0 0 4 3 4: 6750000Hz

Couldn't we just extend spi_divtable to 8 clock speeds, so we could use all the possible speeds on the STM32s without having to make a choice in periph_conf.h?

bergzand commented:
@hugueslarrive I can test this in the next few days. In the meantime could you please remove the obsolete commits here: 9dc703c2fd56bcada931a6c89ccb8815f117e5e9, 12e31589acc48c08e0e5bc2280efad5e4bbbd542 and 95d08ac176a9f533a5bbc3bf1d112bbc134ee9f3. Could you also squash the remaining commits into a single commit with as message "boards/nucleo-f746zg: Add dma, eth, pwm and spi features"

Thanks!

benpicco commented on boards/nucleo-f746zg/include/periph_conf.h:
```diff
@@ -72,23 +131,111 @@ static const uart_conf_t uart_config[] = {
+    }
+};
+
+#define ETH_RX_BUFFER_COUNT (4)
+#define ETH_TX_BUFFER_COUNT (4)
```
dito

hugueslarrive commented on boards/nucleo-f746zg/include/periph_conf.h:
```diff
@@ -72,23 +131,111 @@ static const uart_conf_t uart_config[] = {
+#define ETH_RX_BUFFER_COUNT (4)
+#define ETH_TX_BUFFER_COUNT (4)
+
+#define ETH_RX_BUFFER_SIZE (1524)
+#define ETH_TX_BUFFER_SIZE (1524)
```
I don't know ! That is copy paste from f767zi which is very similar.

@aabadie ?

benpicco replied:
Huh, looks like this got cleaned up in the meantime. 
On `master` I can only find this defined in `cpu/stm32/periph/eth.c`

hugueslarrive replied:
Yes, see #14594 merged 2 days ago, this got cleaned up from f207zg and f767zi.

hugueslarrive commented:
tests/gnrc_udp:
```
2020-08-20 15:50:15,615 # main(): This is RIOT! (Version: 2020.10-devel-879-g28958-boards/nucleo-f746zg)
> ping fe80::ae0a:c0b6:3895:9927
2020-08-20 15:51:39,783 #  ping fe80::ae0a:c0b6:3895:9927
2020-08-20 15:51:39,790 # 12 bytes from fe80::ae0a:c0b6:3895:9927%4: icmp_seq=0 ttl=64 time=0.413 ms
2020-08-20 15:51:40,791 # 12 bytes from fe80::ae0a:c0b6:3895:9927%4: icmp_seq=1 ttl=64 time=0.450 ms
2020-08-20 15:51:41,790 # 12 bytes from fe80::ae0a:c0b6:3895:9927%4: icmp_seq=2 ttl=64 time=0.409 ms
2020-08-20 15:51:41,791 # 
2020-08-20 15:51:41,795 # --- fe80::ae0a:c0b6:3895:9927 PING statistics ---
2020-08-20 15:51:41,800 # 3 packets transmitted, 3 packets received, 0% packet loss
2020-08-20 15:51:41,804 # round-trip min/avg/max = 0.409/0.424/0.450 ms
> ifconfig
2020-08-20 15:52:26,866 #  ifconfig
2020-08-20 15:52:26,869 # Iface  4  HWaddr: 06:17:2A:15:1C:41 
2020-08-20 15:52:26,873 #           L2-PDU:1500  MTU:1500  HL:64  RTR  
2020-08-20 15:52:26,875 #           RTR_ADV  
2020-08-20 15:52:26,878 #           Source address length: 6
2020-08-20 15:52:26,880 #           Link type: wired
2020-08-20 15:52:26,886 #           inet6 addr: fe80::417:2aff:fe15:1c41  scope: link  VAL
2020-08-20 15:52:26,889 #           inet6 group: ff02::2
2020-08-20 15:52:26,892 #           inet6 group: ff02::1
2020-08-20 15:52:26,895 #           inet6 group: ff02::1:ff15:1c41
2020-08-20 15:52:26,896 #           
2020-08-20 15:52:26,899 #           Statistics for Layer 2
2020-08-20 15:52:26,902 #             RX packets 51  bytes 6785
2020-08-20 15:52:26,907 #             TX packets 10 (Multicast: 4)  bytes 752
2020-08-20 15:52:26,910 #             TX succeeded 10 errors 0
2020-08-20 15:52:26,913 #           Statistics for IPv6
2020-08-20 15:52:26,916 #             RX packets 46  bytes 4451
2020-08-20 15:52:26,921 #             TX packets 10 (Multicast: 4)  bytes 612
2020-08-20 15:52:26,924 #             TX succeeded 10 errors 0
2020-08-20 15:52:26,924 # 
> 2020-08-20 15:52:43,836 # Exiting Pyterm
make: *** [/home/hugues/riot-forks/nucleo-f746zg/RIOT/tests/gnrc_udp/../../Makefile.include:741: term] Interrompre
```
flood ping from linux host:
```
# ping -f fe80::417:2aff:fe15:1c41%eno2
PING fe80::417:2aff:fe15:1c41%eno2(fe80::417:2aff:fe15:1c41%eno2) 56 data bytes
.^C
--- fe80::417:2aff:fe15:1c41%eno2 ping statistics ---
9305 packets transmitted, 9304 received, 0,0107469% packet loss, time 5454ms
rtt min/avg/max/mdev = 0.337/0.540/0.966/0.060 ms, ipg/ewma 0.586/0.499 ms
```
Bug described in #13490 is still present.

benpicco commented on boards/nucleo-f746zg/include/periph_conf.h:
```diff
@@ -44,9 +103,9 @@ static const uart_conf_t uart_config[] = {
-#ifdef UART_USE_DMA
-        .dma_stream = 6,
-        .dma_chan   = 4
+#ifdef MODULE_PERIPH_DMA
```
`UART_USE_DMA` is still used by other boards, so I'd keep it.

aabadie replied:
I think `UART_USE_DMA` is a leftover from previous refactoring and it should be `MODULE_PERIPH_DMA`, like it is for most of the other nucleo (and stm32 based boards):

```
$ git grep MODULE_PERIPH_DMA boards/{stm32*,nucleo*,p-*,b-*} | wc -l
72
$ git grep UART_USE_DMA boards/{stm32*,nucleo*,p-*,b-*} | wc -l
14
```

The DMA feature is disabled by default. It's only tested by `tests/periph_dma` and the only way to verify it automatically is to use MODULE_PERIPH_DMA. With UART_USE_DMA, one would have to use CFLAGS.

@hugueslarrive can you post the output of `make BOARD=nucleo-f746zg -C tests/periph_dma flash test` ?

hugueslarrive replied:
All other boards that features periph_dma have  `#ifdef MODULE_PERIPH_DMA` rather than `#ifdef UART_USE_DMA` in their `periph_conf.h`.

```
$ grep -lr UART_USE_DMA
boards/nucleo-l433rc/include/periph_conf.h
boards/nucleo-f303ze/include/periph_conf.h
boards/stm32f769i-disco/include/periph_conf.h
boards/nucleo-f746zg/include/periph_conf.h
boards/stm32l476g-disco/include/periph_conf.h
boards/nucleo-f722ze/include/periph_conf.h
boards/nucleo-l4r5zi/include/periph_conf.h
boards/nucleo-l496zg/include/periph_conf.h
```
There is nothing in the cpu directory to handle UART_USE_DMA rather than MODULE_PERIPH_DMA:
```
$ grep -lr MODULE_PERIPH_DMA
[...]
cpu/stm32/include/periph_cpu.h
cpu/stm32/cpu_init.c
cpu/stm32/periph/uart.c
cpu/stm32/periph/spi.c
cpu/saml21/cpu.c
cpu/saml1x/cpu.c
cpu/sam0_common/include/periph_cpu_common.h
cpu/sam0_common/periph/spi.c
cpu/samd5x/cpu.c
cpu/samd21/cpu.c
```

hugueslarrive commented:
tests/periph_spi:
```
2020-08-20 16:04:40,567 # main(): This is RIOT! (Version: 2020.10-devel-879-g28958-boards/nucleo-f746zg)
2020-08-20 16:04:40,570 # Manual SPI peripheral driver test
2020-08-20 16:04:40,574 # Refer to the README.md file for more information.
2020-08-20 16:04:40,574 # 
2020-08-20 16:04:40,579 # There are 2 SPI devices configured for your platform.
init 0 0 0 0 0
2020-08-20 16:04:55,198 #  init 0 0 0 0 0
2020-08-20 16:04:55,203 # SPI_DEV(0) initialized: mode: 0, clk: 0, cs_port: 0, cs_pin: 0
bench
2020-08-20 16:05:02,778 #  bench
2020-08-20 16:05:02,782 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:05:02,786 # ### Test				Transfer time	user time
2020-08-20 16:05:02,786 # 
2020-08-20 16:05:02,809 #  1 - write 1000 times 1 byte:			19520	19523
2020-08-20 16:05:02,851 #  2 - write 1000 times 2 byte:			38307	38310
2020-08-20 16:05:04,752 #  3 - write 1000 times 100 byte:		1896964	1896967
2020-08-20 16:05:04,796 #  4 - write 1000 times 1 byte to register:	38845	38848
2020-08-20 16:05:04,858 #  5 - write 1000 times 2 byte to register:	57623	57625
2020-08-20 16:05:06,780 #  6 - write 1000 times 100 byte to register:	1916197	1916199
2020-08-20 16:05:06,822 #  7 - read 1000 times 2 byte:			37956	37958
2020-08-20 16:05:08,722 #  8 - read 1000 times 100 byte:		1896418	1896421
2020-08-20 16:05:08,785 #  9 - read 1000 times 2 byte from register:	57558	57561
2020-08-20 16:05:10,706 # 10 - read 1000 times 100 byte from register:	1916040	1916042
2020-08-20 16:05:10,748 # 11 - transfer 1000 times 2 byte:		38146	38148
2020-08-20 16:05:12,650 # 12 - transfer 1000 times 100 byte:		1896599	1896602
2020-08-20 16:05:12,712 # 13 - transfer 1000 times 2 byte to register:	57623	57625
2020-08-20 16:05:14,634 # 14 - transfer 1000 times 100 byte to register:1915975	1915977
2020-08-20 16:05:14,641 # 15 - acquire/release 1000 times:		2989	2992
2020-08-20 16:05:15,643 # -- - SUM:					11786760	11786798
2020-08-20 16:05:15,644 # 
2020-08-20 16:05:15,646 # ### All runs complete ###
init 0 0 1 0 0
2020-08-20 16:05:20,582 #  init 0 0 1 0 0
2020-08-20 16:05:20,587 # SPI_DEV(0) initialized: mode: 0, clk: 1, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:05:21,968 #  bench
2020-08-20 16:05:21,973 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:05:21,976 # ### Test				Transfer time	user time
2020-08-20 16:05:21,976 # 
2020-08-20 16:05:21,999 #  1 - write 1000 times 1 byte:			19520	19523
2020-08-20 16:05:22,041 #  2 - write 1000 times 2 byte:			38307	38310
2020-08-20 16:05:23,943 #  3 - write 1000 times 100 byte:		1896964	1896967
2020-08-20 16:05:23,986 #  4 - write 1000 times 1 byte to register:	38845	38848
2020-08-20 16:05:24,049 #  5 - write 1000 times 2 byte to register:	57623	57625
2020-08-20 16:05:25,971 #  6 - write 1000 times 100 byte to register:	1916197	1916199
2020-08-20 16:05:26,012 #  7 - read 1000 times 2 byte:			37956	37958
2020-08-20 16:05:27,913 #  8 - read 1000 times 100 byte:		1896418	1896421
2020-08-20 16:05:27,976 #  9 - read 1000 times 2 byte from register:	57558	57561
2020-08-20 16:05:29,897 # 10 - read 1000 times 100 byte from register:	1916040	1916042
2020-08-20 16:05:29,939 # 11 - transfer 1000 times 2 byte:		38146	38148
2020-08-20 16:05:31,841 # 12 - transfer 1000 times 100 byte:		1896599	1896602
2020-08-20 16:05:31,903 # 13 - transfer 1000 times 2 byte to register:	57623	57625
2020-08-20 16:05:33,825 # 14 - transfer 1000 times 100 byte to register:1915975	1915977
2020-08-20 16:05:33,832 # 15 - acquire/release 1000 times:		2989	2992
2020-08-20 16:05:34,835 # -- - SUM:					11786760	11786798
2020-08-20 16:05:34,835 # 
2020-08-20 16:05:34,837 # ### All runs complete ###
init 0 0 2 0 0
2020-08-20 16:05:41,905 #  init 0 0 2 0 0
2020-08-20 16:05:41,911 # SPI_DEV(0) initialized: mode: 0, clk: 2, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:05:46,123 #  bench
2020-08-20 16:05:46,127 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:05:46,131 # ### Test				Transfer time	user time
2020-08-20 16:05:46,131 # 
2020-08-20 16:05:46,145 #  1 - write 1000 times 1 byte:			10659	10662
2020-08-20 16:05:46,169 #  2 - write 1000 times 2 byte:			19928	19931
2020-08-20 16:05:47,123 #  3 - write 1000 times 100 byte:		949418	949421
2020-08-20 16:05:47,148 #  4 - write 1000 times 1 byte to register:	21095	21098
2020-08-20 16:05:47,184 #  5 - write 1000 times 2 byte to register:	30401	30403
2020-08-20 16:05:48,148 #  6 - write 1000 times 100 byte to register:	959762	959764
2020-08-20 16:05:48,172 #  7 - read 1000 times 2 byte:			19576	19579
2020-08-20 16:05:49,125 #  8 - read 1000 times 100 byte:		948825	948828
2020-08-20 16:05:49,160 #  9 - read 1000 times 2 byte from register:	30289	30292
2020-08-20 16:05:50,125 # 10 - read 1000 times 100 byte from register:	959558	959561
2020-08-20 16:05:50,148 # 11 - transfer 1000 times 2 byte:		19752	19754
2020-08-20 16:05:51,102 # 12 - transfer 1000 times 100 byte:		948997	949000
2020-08-20 16:05:51,137 # 13 - transfer 1000 times 2 byte to register:	30336	30339
2020-08-20 16:05:52,102 # 14 - transfer 1000 times 100 byte to register:959484	959487
2020-08-20 16:05:52,109 # 15 - acquire/release 1000 times:		2988	2991
2020-08-20 16:05:53,111 # -- - SUM:					5911068	5911110
2020-08-20 16:05:53,112 # 
2020-08-20 16:05:53,114 # ### All runs complete ###
init 0 0 3 0 0
2020-08-20 16:05:57,493 #  init 0 0 3 0 0
2020-08-20 16:05:57,499 # SPI_DEV(0) initialized: mode: 0, clk: 3, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:05:59,287 #  bench
2020-08-20 16:05:59,291 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:05:59,294 # ### Test				Transfer time	user time
2020-08-20 16:05:59,294 # 
2020-08-20 16:05:59,302 #  1 - write 1000 times 1 byte:			3947	3949
2020-08-20 16:05:59,312 #  2 - write 1000 times 2 byte:			6132	6134
2020-08-20 16:05:59,555 #  3 - write 1000 times 100 byte:		238724	238727
2020-08-20 16:05:59,567 #  4 - write 1000 times 1 byte to register:	7716	7718
2020-08-20 16:05:59,582 #  5 - write 1000 times 2 byte to register:	9938	9940
2020-08-20 16:05:59,829 #  6 - write 1000 times 100 byte to register:	242437	242440
2020-08-20 16:05:59,839 #  7 - read 1000 times 2 byte:			5965	5968
2020-08-20 16:06:00,097 #  8 - read 1000 times 100 byte:		253780	253782
2020-08-20 16:06:00,111 #  9 - read 1000 times 2 byte from register:	10012	10014
2020-08-20 16:06:00,374 # 10 - read 1000 times 100 byte from register:	257845	257848
2020-08-20 16:06:00,385 # 11 - transfer 1000 times 2 byte:		6192	6195
2020-08-20 16:06:00,647 # 12 - transfer 1000 times 100 byte:		258099	258102
2020-08-20 16:06:00,662 # 13 - transfer 1000 times 2 byte to register:	10113	10116
2020-08-20 16:06:00,929 # 14 - transfer 1000 times 100 byte to register:261920	261922
2020-08-20 16:06:00,936 # 15 - acquire/release 1000 times:		2988	2991
2020-08-20 16:06:01,939 # -- - SUM:					1575808	1575846
2020-08-20 16:06:01,940 # 
2020-08-20 16:06:01,942 # ### All runs complete ###
init 0 0 4 0 0
2020-08-20 16:06:07,535 #  init 0 0 4 0 0
2020-08-20 16:06:07,540 # SPI_DEV(0) initialized: mode: 0, clk: 4, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:06:09,801 #  bench
2020-08-20 16:06:09,806 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:06:09,809 # ### Test				Transfer time	user time
2020-08-20 16:06:09,809 # 
2020-08-20 16:06:09,815 #  1 - write 1000 times 1 byte:			2872	2875
2020-08-20 16:06:09,823 #  2 - write 1000 times 2 byte:			3863	3866
2020-08-20 16:06:09,948 #  3 - write 1000 times 100 byte:		120298	120301
2020-08-20 16:06:09,958 #  4 - write 1000 times 1 byte to register:	5530	5532
2020-08-20 16:06:09,968 #  5 - write 1000 times 2 byte to register:	6512	6515
2020-08-20 16:06:10,097 #  6 - write 1000 times 100 byte to register:	122855	122857
2020-08-20 16:06:10,104 #  7 - read 1000 times 2 byte:			3743	3746
2020-08-20 16:06:10,251 #  8 - read 1000 times 100 byte:		142669	142671
2020-08-20 16:06:10,262 #  9 - read 1000 times 2 byte from register:	6679	6681
2020-08-20 16:06:10,413 # 10 - read 1000 times 100 byte from register:	145623	145625
2020-08-20 16:06:10,420 # 11 - transfer 1000 times 2 byte:		4043	4047
2020-08-20 16:06:10,576 # 12 - transfer 1000 times 100 byte:		150691	150694
2020-08-20 16:06:10,587 # 13 - transfer 1000 times 2 byte to register:	6854	6857
2020-08-20 16:06:10,746 # 14 - transfer 1000 times 100 byte to register:153401	153403
2020-08-20 16:06:10,753 # 15 - acquire/release 1000 times:		2988	2991
2020-08-20 16:06:11,756 # -- - SUM:					878621	878661
2020-08-20 16:06:11,756 # 
2020-08-20 16:06:11,758 # ### All runs complete ###
init 1 0 0 0 0
2020-08-20 16:06:23,331 #  init 1 0 0 0 0
2020-08-20 16:06:23,336 # SPI_DEV(1) initialized: mode: 0, clk: 0, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:06:25,408 #  bench
2020-08-20 16:06:25,413 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:06:25,416 # ### Test				Transfer time	user time
2020-08-20 16:06:25,416 # 
2020-08-20 16:06:25,439 #  1 - write 1000 times 1 byte:			19521	19522
2020-08-20 16:06:25,481 #  2 - write 1000 times 2 byte:			38308	38310
2020-08-20 16:06:27,383 #  3 - write 1000 times 100 byte:		1896965	1896968
2020-08-20 16:06:27,426 #  4 - write 1000 times 1 byte to register:	38845	38848
2020-08-20 16:06:27,489 #  5 - write 1000 times 2 byte to register:	57623	57626
2020-08-20 16:06:29,410 #  6 - write 1000 times 100 byte to register:	1916197	1916199
2020-08-20 16:06:29,452 #  7 - read 1000 times 2 byte:			37955	37958
2020-08-20 16:06:31,353 #  8 - read 1000 times 100 byte:		1896419	1896421
2020-08-20 16:06:31,415 #  9 - read 1000 times 2 byte from register:	57558	57561
2020-08-20 16:06:33,336 # 10 - read 1000 times 100 byte from register:	1916039	1916042
2020-08-20 16:06:33,379 # 11 - transfer 1000 times 2 byte:		38145	38148
2020-08-20 16:06:35,280 # 12 - transfer 1000 times 100 byte:		1896599	1896602
2020-08-20 16:06:35,342 # 13 - transfer 1000 times 2 byte to register:	57623	57625
2020-08-20 16:06:37,264 # 14 - transfer 1000 times 100 byte to register:1915975	1915978
2020-08-20 16:06:37,271 # 15 - acquire/release 1000 times:		2989	2991
2020-08-20 16:06:38,274 # -- - SUM:					11786761	11786799
2020-08-20 16:06:38,274 # 
2020-08-20 16:06:38,276 # ### All runs complete ###
init 1 0 1 0 0
2020-08-20 16:06:42,532 #  init 1 0 1 0 0
2020-08-20 16:06:42,538 # SPI_DEV(1) initialized: mode: 0, clk: 1, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:06:45,491 #  bench
2020-08-20 16:06:45,495 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:06:45,498 # ### Test				Transfer time	user time
2020-08-20 16:06:45,499 # 
2020-08-20 16:06:45,521 #  1 - write 1000 times 1 byte:			19520	19523
2020-08-20 16:06:45,564 #  2 - write 1000 times 2 byte:			38307	38310
2020-08-20 16:06:47,466 #  3 - write 1000 times 100 byte:		1896964	1896967
2020-08-20 16:06:47,509 #  4 - write 1000 times 1 byte to register:	38845	38848
2020-08-20 16:06:47,572 #  5 - write 1000 times 2 byte to register:	57623	57625
2020-08-20 16:06:49,493 #  6 - write 1000 times 100 byte to register:	1916197	1916199
2020-08-20 16:06:49,534 #  7 - read 1000 times 2 byte:			37956	37958
2020-08-20 16:06:51,435 #  8 - read 1000 times 100 byte:		1896418	1896421
2020-08-20 16:06:51,498 #  9 - read 1000 times 2 byte from register:	57558	57561
2020-08-20 16:06:53,420 # 10 - read 1000 times 100 byte from register:	1916040	1916042
2020-08-20 16:06:53,462 # 11 - transfer 1000 times 2 byte:		38146	38148
2020-08-20 16:06:55,363 # 12 - transfer 1000 times 100 byte:		1896599	1896602
2020-08-20 16:06:55,426 # 13 - transfer 1000 times 2 byte to register:	57623	57625
2020-08-20 16:06:57,347 # 14 - transfer 1000 times 100 byte to register:1915975	1915977
2020-08-20 16:06:57,354 # 15 - acquire/release 1000 times:		2989	2992
2020-08-20 16:06:58,357 # -- - SUM:					11786760	11786798
2020-08-20 16:06:58,357 # 
2020-08-20 16:06:58,359 # ### All runs complete ###
init 1 0 2 0 0
2020-08-20 16:07:03,536 #  init 1 0 2 0 0
2020-08-20 16:07:03,542 # SPI_DEV(1) initialized: mode: 0, clk: 2, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:07:06,792 #  bench
2020-08-20 16:07:06,796 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:07:06,799 # ### Test				Transfer time	user time
2020-08-20 16:07:06,799 # 
2020-08-20 16:07:06,814 #  1 - write 1000 times 1 byte:			10660	10662
2020-08-20 16:07:06,838 #  2 - write 1000 times 2 byte:			19927	19930
2020-08-20 16:07:07,792 #  3 - write 1000 times 100 byte:		949419	949421
2020-08-20 16:07:07,817 #  4 - write 1000 times 1 byte to register:	21095	21097
2020-08-20 16:07:07,852 #  5 - write 1000 times 2 byte to register:	30401	30404
2020-08-20 16:07:08,817 #  6 - write 1000 times 100 byte to register:	959762	959764
2020-08-20 16:07:08,841 #  7 - read 1000 times 2 byte:			19576	19578
2020-08-20 16:07:09,794 #  8 - read 1000 times 100 byte:		948826	948828
2020-08-20 16:07:09,829 #  9 - read 1000 times 2 byte from register:	30290	30292
2020-08-20 16:07:10,793 # 10 - read 1000 times 100 byte from register:	959558	959561
2020-08-20 16:07:10,817 # 11 - transfer 1000 times 2 byte:		19752	19755
2020-08-20 16:07:11,771 # 12 - transfer 1000 times 100 byte:		948998	949000
2020-08-20 16:07:11,806 # 13 - transfer 1000 times 2 byte to register:	30336	30338
2020-08-20 16:07:12,771 # 14 - transfer 1000 times 100 byte to register:959485	959487
2020-08-20 16:07:12,778 # 15 - acquire/release 1000 times:		2989	2991
2020-08-20 16:07:13,780 # -- - SUM:					5911074	5911108
2020-08-20 16:07:13,781 # 
2020-08-20 16:07:13,783 # ### All runs complete ###
init 1 0 3 0 0
2020-08-20 16:07:18,436 #  init 1 0 3 0 0
2020-08-20 16:07:18,442 # SPI_DEV(1) initialized: mode: 0, clk: 3, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:07:20,363 #  bench
2020-08-20 16:07:20,367 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:07:20,370 # ### Test				Transfer time	user time
2020-08-20 16:07:20,371 # 
2020-08-20 16:07:20,378 #  1 - write 1000 times 1 byte:			3947	3949
2020-08-20 16:07:20,388 #  2 - write 1000 times 2 byte:			6131	6134
2020-08-20 16:07:20,631 #  3 - write 1000 times 100 byte:		238724	238726
2020-08-20 16:07:20,643 #  4 - write 1000 times 1 byte to register:	7715	7718
2020-08-20 16:07:20,657 #  5 - write 1000 times 2 byte to register:	9937	9940
2020-08-20 16:07:20,905 #  6 - write 1000 times 100 byte to register:	242438	242441
2020-08-20 16:07:20,915 #  7 - read 1000 times 2 byte:			5965	5968
2020-08-20 16:07:21,172 #  8 - read 1000 times 100 byte:		253779	253782
2020-08-20 16:07:21,187 #  9 - read 1000 times 2 byte from register:	10012	10014
2020-08-20 16:07:21,450 # 10 - read 1000 times 100 byte from register:	257845	257847
2020-08-20 16:07:21,460 # 11 - transfer 1000 times 2 byte:		6192	6195
2020-08-20 16:07:21,723 # 12 - transfer 1000 times 100 byte:		258099	258102
2020-08-20 16:07:21,738 # 13 - transfer 1000 times 2 byte to register:	10114	10117
2020-08-20 16:07:22,005 # 14 - transfer 1000 times 100 byte to register:261919	261922
2020-08-20 16:07:22,012 # 15 - acquire/release 1000 times:		2988	2992
2020-08-20 16:07:23,015 # -- - SUM:					1575805	1575847
2020-08-20 16:07:23,015 # 
2020-08-20 16:07:23,017 # ### All runs complete ###
init 1 0 4 0 0
2020-08-20 16:07:31,843 #  init 1 0 4 0 0
2020-08-20 16:07:31,848 # SPI_DEV(1) initialized: mode: 0, clk: 4, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:07:34,050 #  bench
2020-08-20 16:07:34,055 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:07:34,058 # ### Test				Transfer time	user time
2020-08-20 16:07:34,059 # 
2020-08-20 16:07:34,065 #  1 - write 1000 times 1 byte:			2873	2875
2020-08-20 16:07:34,073 #  2 - write 1000 times 2 byte:			3863	3865
2020-08-20 16:07:34,197 #  3 - write 1000 times 100 byte:		120298	120300
2020-08-20 16:07:34,207 #  4 - write 1000 times 1 byte to register:	5530	5533
2020-08-20 16:07:34,219 #  5 - write 1000 times 2 byte to register:	6511	6514
2020-08-20 16:07:34,346 #  6 - write 1000 times 100 byte to register:	122854	122857
2020-08-20 16:07:34,354 #  7 - read 1000 times 2 byte:			3742	3745
2020-08-20 16:07:34,500 #  8 - read 1000 times 100 byte:		142669	142671
2020-08-20 16:07:34,512 #  9 - read 1000 times 2 byte from register:	6678	6681
2020-08-20 16:07:34,663 # 10 - read 1000 times 100 byte from register:	145623	145626
2020-08-20 16:07:34,671 # 11 - transfer 1000 times 2 byte:		4044	4046
2020-08-20 16:07:34,826 # 12 - transfer 1000 times 100 byte:		150692	150694
2020-08-20 16:07:34,837 # 13 - transfer 1000 times 2 byte to register:	6854	6857
2020-08-20 16:07:34,996 # 14 - transfer 1000 times 100 byte to register:153401	153404
2020-08-20 16:07:35,003 # 15 - acquire/release 1000 times:		2988	2991
2020-08-20 16:07:36,005 # -- - SUM:					878620	878659
2020-08-20 16:07:36,006 # 
2020-08-20 16:07:36,008 # ### All runs complete ###
> 
```

hugueslarrive commented:
tests/periph_spi_dma:
```
2020-08-20 16:18:16,714 # main(): This is RIOT! (Version: 2020.10-devel-879-g28958-boards/nucleo-f746zg)
2020-08-20 16:18:16,717 # Manual SPI peripheral driver test
2020-08-20 16:18:16,721 # Refer to the README.md file for more information.
2020-08-20 16:18:16,722 # 
2020-08-20 16:18:16,726 # There are 2 SPI devices configured for your platform.
init 0 0 0 0 0
2020-08-20 16:18:20,617 #  init 0 0 0 0 0
2020-08-20 16:18:20,623 # SPI_DEV(0) initialized: mode: 0, clk: 0, cs_port: 0, cs_pin: 0
bench
2020-08-20 16:18:28,038 #  bench
2020-08-20 16:18:28,043 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:18:28,046 # ### Test				Transfer time	user time
2020-08-20 16:18:28,046 # 
2020-08-20 16:18:28,079 #  1 - write 1000 times 1 byte:			29538	12133
2020-08-20 16:18:28,131 #  2 - write 1000 times 2 byte:			48372	12299
2020-08-20 16:18:30,043 #  3 - write 1000 times 100 byte:		1906556	10226
2020-08-20 16:18:30,106 #  4 - write 1000 times 1 byte to register:	58705	24224
2020-08-20 16:18:30,189 #  5 - write 1000 times 2 byte to register:	77927	24597
2020-08-20 16:18:32,129 #  6 - write 1000 times 100 byte to register:	1935890	22226
2020-08-20 16:18:32,181 #  7 - read 1000 times 2 byte:			48001	12004
2020-08-20 16:18:34,092 #  8 - read 1000 times 100 byte:		1906241	9708
2020-08-20 16:18:34,174 #  9 - read 1000 times 2 byte from register:	77446	24224
2020-08-20 16:18:36,115 # 10 - read 1000 times 100 byte from register:	1935649	21780
2020-08-20 16:18:36,167 # 11 - transfer 1000 times 2 byte:		48094	11966
2020-08-20 16:18:38,078 # 12 - transfer 1000 times 100 byte:		1906436	9911
2020-08-20 16:18:38,160 # 13 - transfer 1000 times 2 byte to register:	77557	24225
2020-08-20 16:18:40,101 # 14 - transfer 1000 times 100 byte to register:1935594	21967
2020-08-20 16:18:40,115 # 15 - acquire/release 1000 times:		9794	9797
2020-08-20 16:18:41,118 # -- - SUM:					12001800	251287
2020-08-20 16:18:41,118 # 
2020-08-20 16:18:41,120 # ### All runs complete ###
init 0 0 1 0 0
2020-08-20 16:18:58,150 #  init 0 0 1 0 0
2020-08-20 16:18:58,156 # SPI_DEV(0) initialized: mode: 0, clk: 1, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:19:00,543 #  bench
2020-08-20 16:19:00,548 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:19:00,551 # ### Test				Transfer time	user time
2020-08-20 16:19:00,551 # 
2020-08-20 16:19:00,585 #  1 - write 1000 times 1 byte:			29538	12133
2020-08-20 16:19:00,637 #  2 - write 1000 times 2 byte:			48372	12300
2020-08-20 16:19:02,548 #  3 - write 1000 times 100 byte:		1906557	10226
2020-08-20 16:19:02,611 #  4 - write 1000 times 1 byte to register:	58704	24227
2020-08-20 16:19:02,694 #  5 - write 1000 times 2 byte to register:	77927	24596
2020-08-20 16:19:04,635 #  6 - write 1000 times 100 byte to register:	1935890	22226
2020-08-20 16:19:04,687 #  7 - read 1000 times 2 byte:			48002	12003
2020-08-20 16:19:06,597 #  8 - read 1000 times 100 byte:		1906242	9706
2020-08-20 16:19:06,679 #  9 - read 1000 times 2 byte from register:	77445	24226
2020-08-20 16:19:08,620 # 10 - read 1000 times 100 byte from register:	1935649	21782
2020-08-20 16:19:08,672 # 11 - transfer 1000 times 2 byte:		48094	11968
2020-08-20 16:19:10,583 # 12 - transfer 1000 times 100 byte:		1906436	9912
2020-08-20 16:19:10,665 # 13 - transfer 1000 times 2 byte to register:	77557	24225
2020-08-20 16:19:12,607 # 14 - transfer 1000 times 100 byte to register:1935594	21967
2020-08-20 16:19:12,620 # 15 - acquire/release 1000 times:		9794	9797
2020-08-20 16:19:13,623 # -- - SUM:					12001801	251294
2020-08-20 16:19:13,623 # 
2020-08-20 16:19:13,625 # ### All runs complete ###
init 0 0 2 0 0
2020-08-20 16:19:19,111 #  init 0 0 2 0 0
2020-08-20 16:19:19,117 # SPI_DEV(0) initialized: mode: 0, clk: 2, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:19:21,379 #  bench
2020-08-20 16:19:21,384 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:19:21,387 # ### Test				Transfer time	user time
2020-08-20 16:19:21,387 # 
2020-08-20 16:19:21,412 #  1 - write 1000 times 1 byte:			20649	12131
2020-08-20 16:19:21,446 #  2 - write 1000 times 2 byte:			30001	13003
2020-08-20 16:19:22,409 #  3 - write 1000 times 100 byte:		959001	11002
2020-08-20 16:19:22,454 #  4 - write 1000 times 1 byte to register:	40927	24225
2020-08-20 16:19:22,510 #  5 - write 1000 times 2 byte to register:	50668	24670
2020-08-20 16:19:23,494 #  6 - write 1000 times 100 byte to register:	979445	22226
2020-08-20 16:19:23,528 #  7 - read 1000 times 2 byte:			29631	11893
2020-08-20 16:19:24,490 #  8 - read 1000 times 100 byte:		958686	9707
2020-08-20 16:19:24,545 #  9 - read 1000 times 2 byte from register:	50187	24152
2020-08-20 16:19:25,529 # 10 - read 1000 times 100 byte from register:	979205	21778
2020-08-20 16:19:25,563 # 11 - transfer 1000 times 2 byte:		29723	12004
2020-08-20 16:19:26,526 # 12 - transfer 1000 times 100 byte:		958881	9912
2020-08-20 16:19:26,582 # 13 - transfer 1000 times 2 byte to register:	50297	24299
2020-08-20 16:19:27,566 # 14 - transfer 1000 times 100 byte to register:979149	21966
2020-08-20 16:19:27,580 # 15 - acquire/release 1000 times:		9794	9797
2020-08-20 16:19:28,583 # -- - SUM:					6126244	252765
2020-08-20 16:19:28,583 # 
2020-08-20 16:19:28,586 # ### All runs complete ###
init 0 0 3 0 0
2020-08-20 16:19:34,096 #  init 0 0 3 0 0
2020-08-20 16:19:34,102 # SPI_DEV(0) initialized: mode: 0, clk: 3, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:19:35,902 #  bench
2020-08-20 16:19:35,906 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:19:35,910 # ### Test				Transfer time	user time
2020-08-20 16:19:35,910 # 
2020-08-20 16:19:35,926 #  1 - write 1000 times 1 byte:			12594	12597
2020-08-20 16:19:35,948 #  2 - write 1000 times 2 byte:			18335	12337
2020-08-20 16:19:36,201 #  3 - write 1000 times 100 byte:		248335	10004
2020-08-20 16:19:36,230 #  4 - write 1000 times 1 byte to register:	24797	24801
2020-08-20 16:19:36,266 #  5 - write 1000 times 2 byte to register:	31075	25078
2020-08-20 16:19:36,532 #  6 - write 1000 times 100 byte to register:	260853	22892
2020-08-20 16:19:36,554 #  7 - read 1000 times 2 byte:			17908	11892
2020-08-20 16:19:36,806 #  8 - read 1000 times 100 byte:		248019	9716
2020-08-20 16:19:36,841 #  9 - read 1000 times 2 byte from register:	30760	24763
2020-08-20 16:19:37,107 # 10 - read 1000 times 100 byte from register:	260816	22819
2020-08-20 16:19:37,129 # 11 - transfer 1000 times 2 byte:		17983	11965
2020-08-20 16:19:37,382 # 12 - transfer 1000 times 100 byte:		248214	9912
2020-08-20 16:19:37,418 # 13 - transfer 1000 times 2 byte to register:	30890	24782
2020-08-20 16:19:37,683 # 14 - transfer 1000 times 100 byte to register:260650	22707
2020-08-20 16:19:37,697 # 15 - acquire/release 1000 times:		9794	9797
2020-08-20 16:19:38,700 # -- - SUM:					1721023	256062
2020-08-20 16:19:38,701 # 
2020-08-20 16:19:38,704 # ### All runs complete ###
init 0 0 4 0 0
2020-08-20 16:19:43,875 #  init 0 0 4 0 0
2020-08-20 16:19:43,881 # SPI_DEV(0) initialized: mode: 0, clk: 4, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:19:45,405 #  bench
2020-08-20 16:19:45,409 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:19:45,412 # ### Test				Transfer time	user time
2020-08-20 16:19:45,413 # 
2020-08-20 16:19:45,424 #  1 - write 1000 times 1 byte:			7889	7892
2020-08-20 16:19:45,441 #  2 - write 1000 times 2 byte:			12594	12597
2020-08-20 16:19:45,575 #  3 - write 1000 times 100 byte:		129890	10226
2020-08-20 16:19:45,595 #  4 - write 1000 times 1 byte to register:	15270	15274
2020-08-20 16:19:45,620 #  5 - write 1000 times 2 byte to register:	20029	20033
2020-08-20 16:19:45,763 #  6 - write 1000 times 100 byte to register:	137946	18334
2020-08-20 16:19:45,778 #  7 - read 1000 times 2 byte:			12141	12143
2020-08-20 16:19:45,912 #  8 - read 1000 times 100 byte:		129575	9707
2020-08-20 16:19:45,937 #  9 - read 1000 times 2 byte from register:	19798	19801
2020-08-20 16:19:46,080 # 10 - read 1000 times 100 byte from register:	137853	18449
2020-08-20 16:19:46,096 # 11 - transfer 1000 times 2 byte:		12409	12411
2020-08-20 16:19:46,230 # 12 - transfer 1000 times 100 byte:		129769	9912
2020-08-20 16:19:46,255 # 13 - transfer 1000 times 2 byte to register:	19835	19838
2020-08-20 16:19:46,398 # 14 - transfer 1000 times 100 byte to register:137854	18263
2020-08-20 16:19:46,411 # 15 - acquire/release 1000 times:		9794	9797
2020-08-20 16:19:47,414 # -- - SUM:					932646	214677
2020-08-20 16:19:47,415 # 
2020-08-20 16:19:47,416 # ### All runs complete ###
init 1 0 0 0 0
2020-08-20 16:19:59,346 #  init 1 0 0 0 0
2020-08-20 16:19:59,351 # SPI_DEV(1) initialized: mode: 0, clk: 0, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:20:01,604 #  bench
2020-08-20 16:20:01,609 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:20:01,612 # ### Test				Transfer time	user time
2020-08-20 16:20:01,612 # 
2020-08-20 16:20:01,646 #  1 - write 1000 times 1 byte:			29724	12058
2020-08-20 16:20:01,698 #  2 - write 1000 times 2 byte:			48428	12207
2020-08-20 16:20:03,609 #  3 - write 1000 times 100 byte:		1906612	10171
2020-08-20 16:20:03,672 #  4 - write 1000 times 1 byte to register:	58853	24114
2020-08-20 16:20:03,755 #  5 - write 1000 times 2 byte to register:	78038	24410
2020-08-20 16:20:05,696 #  6 - write 1000 times 100 byte to register:	1936002	22004
2020-08-20 16:20:05,748 #  7 - read 1000 times 2 byte:			48057	11781
2020-08-20 16:20:07,658 #  8 - read 1000 times 100 byte:		1906297	9708
2020-08-20 16:20:07,741 #  9 - read 1000 times 2 byte from register:	77575	23985
2020-08-20 16:20:09,682 # 10 - read 1000 times 100 byte from register:	1935630	21820
2020-08-20 16:20:09,734 # 11 - transfer 1000 times 2 byte:		48149	11892
2020-08-20 16:20:11,645 # 12 - transfer 1000 times 100 byte:		1906501	10003
2020-08-20 16:20:11,727 # 13 - transfer 1000 times 2 byte to register:	77667	24004
2020-08-20 16:20:13,669 # 14 - transfer 1000 times 100 byte to register:1935909	21869
2020-08-20 16:20:13,682 # 15 - acquire/release 1000 times:		9780	9783
2020-08-20 16:20:14,685 # -- - SUM:					12003222	249809
2020-08-20 16:20:14,685 # 
2020-08-20 16:20:14,687 # ### All runs complete ###
init 1 0 1 0 0
2020-08-20 16:20:27,038 #  init 1 0 1 0 0
2020-08-20 16:20:27,044 # SPI_DEV(1) initialized: mode: 0, clk: 1, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:20:28,688 #  bench
2020-08-20 16:20:28,693 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:20:28,696 # ### Test				Transfer time	user time
2020-08-20 16:20:28,696 # 
2020-08-20 16:20:28,730 #  1 - write 1000 times 1 byte:			29723	12058
2020-08-20 16:20:28,782 #  2 - write 1000 times 2 byte:			48427	12206
2020-08-20 16:20:30,693 #  3 - write 1000 times 100 byte:		1906612	10171
2020-08-20 16:20:30,756 #  4 - write 1000 times 1 byte to register:	58853	24116
2020-08-20 16:20:30,839 #  5 - write 1000 times 2 byte to register:	78038	24410
2020-08-20 16:20:32,780 #  6 - write 1000 times 100 byte to register:	1936001	22003
2020-08-20 16:20:32,832 #  7 - read 1000 times 2 byte:			48056	11783
2020-08-20 16:20:34,742 #  8 - read 1000 times 100 byte:		1906297	9708
2020-08-20 16:20:34,825 #  9 - read 1000 times 2 byte from register:	77575	23983
2020-08-20 16:20:36,766 # 10 - read 1000 times 100 byte from register:	1935631	21818
2020-08-20 16:20:36,818 # 11 - transfer 1000 times 2 byte:		48150	11892
2020-08-20 16:20:38,729 # 12 - transfer 1000 times 100 byte:		1906501	10004
2020-08-20 16:20:38,811 # 13 - transfer 1000 times 2 byte to register:	77668	24003
2020-08-20 16:20:40,753 # 14 - transfer 1000 times 100 byte to register:1935909	21874
2020-08-20 16:20:40,766 # 15 - acquire/release 1000 times:		9780	9783
2020-08-20 16:20:41,769 # -- - SUM:					12003221	249812
2020-08-20 16:20:41,769 # 
2020-08-20 16:20:41,771 # ### All runs complete ###
init 1 0 2 0 0
2020-08-20 16:20:48,794 #  init 1 0 2 0 0
2020-08-20 16:20:48,801 # SPI_DEV(1) initialized: mode: 0, clk: 2, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:20:50,522 #  bench
2020-08-20 16:20:50,527 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:20:50,531 # ### Test				Transfer time	user time
2020-08-20 16:20:50,531 # 
2020-08-20 16:20:50,555 #  1 - write 1000 times 1 byte:			20835	12169
2020-08-20 16:20:50,589 #  2 - write 1000 times 2 byte:			30056	12169
2020-08-20 16:20:51,552 #  3 - write 1000 times 100 byte:		959057	10170
2020-08-20 16:20:51,598 #  4 - write 1000 times 1 byte to register:	41076	24042
2020-08-20 16:20:51,653 #  5 - write 1000 times 2 byte to register:	50779	24447
2020-08-20 16:20:52,638 #  6 - write 1000 times 100 byte to register:	979556	22226
2020-08-20 16:20:52,671 #  7 - read 1000 times 2 byte:			29686	11800
2020-08-20 16:20:53,634 #  8 - read 1000 times 100 byte:		958741	9708
2020-08-20 16:20:53,689 #  9 - read 1000 times 2 byte from register:	50316	23983
2020-08-20 16:20:54,674 # 10 - read 1000 times 100 byte from register:	979186	21782
2020-08-20 16:20:54,708 # 11 - transfer 1000 times 2 byte:		29779	11893
2020-08-20 16:20:55,671 # 12 - transfer 1000 times 100 byte:		958945	9948
2020-08-20 16:20:55,727 # 13 - transfer 1000 times 2 byte to register:	50409	24114
2020-08-20 16:20:56,711 # 14 - transfer 1000 times 100 byte to register:979464	21876
2020-08-20 16:20:56,725 # 15 - acquire/release 1000 times:		9780	9783
2020-08-20 16:20:57,728 # -- - SUM:					6127665	250110
2020-08-20 16:20:57,728 # 
2020-08-20 16:20:57,730 # ### All runs complete ###
init 1 0 3 0 0
2020-08-20 16:21:02,445 #  init 1 0 3 0 0
2020-08-20 16:21:02,451 # SPI_DEV(1) initialized: mode: 0, clk: 3, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:21:04,817 #  bench
2020-08-20 16:21:04,822 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:21:04,825 # ### Test				Transfer time	user time
2020-08-20 16:21:04,826 # 
2020-08-20 16:21:04,842 #  1 - write 1000 times 1 byte:			12302	12306
2020-08-20 16:21:04,863 #  2 - write 1000 times 2 byte:			18297	12225
2020-08-20 16:21:05,116 #  3 - write 1000 times 100 byte:		248390	10171
2020-08-20 16:21:05,146 #  4 - write 1000 times 1 byte to register:	24594	24597
2020-08-20 16:21:05,181 #  5 - write 1000 times 2 byte to register:	30687	24616
2020-08-20 16:21:05,447 #  6 - write 1000 times 100 byte to register:	260798	22800
2020-08-20 16:21:05,469 #  7 - read 1000 times 2 byte:			17872	11800
2020-08-20 16:21:05,721 #  8 - read 1000 times 100 byte:		248075	9707
2020-08-20 16:21:05,756 #  9 - read 1000 times 2 byte from register:	30557	24504
2020-08-20 16:21:06,022 # 10 - read 1000 times 100 byte from register:	260741	22651
2020-08-20 16:21:06,044 # 11 - transfer 1000 times 2 byte:		17946	11893
2020-08-20 16:21:06,296 # 12 - transfer 1000 times 100 byte:		248279	9893
2020-08-20 16:21:06,332 # 13 - transfer 1000 times 2 byte to register:	30594	24522
2020-08-20 16:21:06,598 # 14 - transfer 1000 times 100 byte to register:260483	22486
2020-08-20 16:21:06,611 # 15 - acquire/release 1000 times:		9780	9783
2020-08-20 16:21:07,613 # -- - SUM:					1719395	253954
2020-08-20 16:21:07,614 # 
2020-08-20 16:21:07,617 # ### All runs complete ###
init 1 0 4 0 0
2020-08-20 16:21:12,501 #  init 1 0 4 0 0
2020-08-20 16:21:12,507 # SPI_DEV(1) initialized: mode: 0, clk: 4, cs_port: 0, cs_pin: 0
bench 0
2020-08-20 16:21:14,389 #  bench
2020-08-20 16:21:14,394 # ### Running some benchmarks, all values in [us] ###
2020-08-20 16:21:14,397 # ### Test				Transfer time	user time
2020-08-20 16:21:14,397 # 
2020-08-20 16:21:14,408 #  1 - write 1000 times 1 byte:			7456	7458
2020-08-20 16:21:14,424 #  2 - write 1000 times 2 byte:			12316	12319
2020-08-20 16:21:14,558 #  3 - write 1000 times 100 byte:		129945	10172
2020-08-20 16:21:14,578 #  4 - write 1000 times 1 byte to register:	15113	15116
2020-08-20 16:21:14,603 #  5 - write 1000 times 2 byte to register:	20011	20013
2020-08-20 16:21:14,745 #  6 - write 1000 times 100 byte to register:	137557	18115
2020-08-20 16:21:14,761 #  7 - read 1000 times 2 byte:			11839	11843
2020-08-20 16:21:14,895 #  8 - read 1000 times 100 byte:		129630	9708
2020-08-20 16:21:14,920 #  9 - read 1000 times 2 byte from register:	20205	20207
2020-08-20 16:21:15,063 # 10 - read 1000 times 100 byte from register:	138088	18546
2020-08-20 16:21:15,079 # 11 - transfer 1000 times 2 byte:		12159	12162
2020-08-20 16:21:15,213 # 12 - transfer 1000 times 100 byte:		129835	9837
2020-08-20 16:21:15,238 # 13 - transfer 1000 times 2 byte to register:	19548	19551
2020-08-20 16:21:15,381 # 14 - transfer 1000 times 100 byte to register:137548	17957
2020-08-20 16:21:15,394 # 15 - acquire/release 1000 times:		9780	9783
2020-08-20 16:21:16,397 # -- - SUM:					931030	212787
2020-08-20 16:21:16,397 # 
2020-08-20 16:21:16,399 # ### All runs complete ###
> 
```

hugueslarrive commented:
tests/periph_pwm:
```2020-08-20 16:33:45,629 # main(): This is RIOT! (Version: 2020.10-devel-879-g28958-boards/nucleo-f746zg)
2020-08-20 16:33:45,631 # PWM peripheral driver test
2020-08-20 16:33:45,631 # 
> osci
2020-08-20 16:33:50,651 #  osci
2020-08-20 16:33:50,651 # 
2020-08-20 16:33:50,652 # RIOT PWM test
2020-08-20 16:33:50,657 # Connect an LED or scope to PWM pins to see something.
2020-08-20 16:33:50,657 # 
2020-08-20 16:33:50,660 # Available PWM device between 0 and 1
2020-08-20 16:33:50,663 # Initialized PWM_0 @ 1000Hz.
2020-08-20 16:33:50,665 # Initialized PWM_1 @ 1002Hz.
2020-08-20 16:33:50,665 # 
2020-08-20 16:33:50,669 # Letting the PWM pins oscillate now...
```

![osci](https://user-images.githubusercontent.com/67432403/90786074-6eb53d80-e303-11ea-8376-c41f64579f2c.png)
